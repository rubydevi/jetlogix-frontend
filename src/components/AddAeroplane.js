import React, { useState, useContext } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { createAeroplane } from '../redux/aeroplanes/aeroplanesActions';
import AuthContext from '../context/AuthProvider';
import AddAeroplaneForm from './AddAeroplaneForm';
import Toast from './Toast';

const AddAeroplane = () => {
  const { auth } = useContext(AuthContext);
  const [aeroplane, setAeroplane] = useState({
    name: '',
    model: '',
    image: '',
    description: '',
    number_of_seats: '',
    location: '',
    fee: 0,
    reserved: false,
  });

  const dispatch = useDispatch();
  const history = useNavigate();
  const [error, setError] = useState(null);

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const response = await dispatch(createAeroplane({ aeroplane, auth }));
      if (response.payload && !response.error) {
        history('/aeroplanes');
      } else {
        setError('Something went wrong');
        const liveToast = new window.bootstrap.Toast(document.getElementById('liveToast'));
        liveToast.show();
      }
    } catch (error) {
      setError(`Error: ${error.message}`);
    }
  };

  return (
    <div className="container">
      <div className="col-12 text-center">
        <h3 className="fw-normal"> Add a New Aeroplane</h3>
      </div>
      <Toast error={error} />
      <AddAeroplaneForm
        aeroplane={aeroplane}
        setAeroplane={setAeroplane}
        handleSubmit={handleSubmit}
      />
    </div>
  );
};

export default AddAeroplane;
