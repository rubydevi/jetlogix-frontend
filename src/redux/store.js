import { configureStore } from '@reduxjs/toolkit';
import aeroplanesReducer from './aeroplanes/aeroplanesSlice';
import reservationsReducer from './reservations/reservationSlice';

const store = configureStore({
  reducer: {
    aeroplanes: aeroplanesReducer,
    reservation: reservationsReducer,
  },
});

export default store;
