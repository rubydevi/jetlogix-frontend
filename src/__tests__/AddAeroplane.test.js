import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import AddAeroplane from '../components/AddAeroplane';

test('renders AddAeroplane component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <AddAeroplane />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
