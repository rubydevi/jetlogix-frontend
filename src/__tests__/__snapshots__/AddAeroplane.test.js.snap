// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddAeroplane component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="col-12 text-center"
    >
      <h3
        class="fw-normal"
      >
         Add a New Aeroplane
      </h3>
    </div>
    <div
      class="toast-container position-fixed top-0 end-0 p-3"
    >
      <div
        aria-atomic="true"
        aria-live="assertive"
        class="toast"
        id="liveToast"
        role="alert"
      >
        <div
          class="toast-header"
        >
          <div
            class="float-sm-start"
          >
            <svg
              class="me-2"
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4c-4.879 0-9 4.121-9 9s4.121 9 9 9 9-4.121 9-9-4.121-9-9-9zm0 16c-3.794 0-7-3.206-7-7s3.206-7 7-7 7 3.206 7 7-3.206 7-7 7z"
              />
              <path
                d="M13 12V8h-2v6h6v-2zm4.284-8.293 1.412-1.416 3.01 3-1.413 1.417zm-10.586 0-2.99 2.999L2.29 5.294l2.99-3z"
              />
            </svg>
            <i
              class="bi bi-react me-2"
            />
          </div>
          <strong
            class="me-auto"
          >
            Alert
          </strong>
          <div
            class="float-sm-end"
          >
            <button
              aria-label="Close"
              class="btn-close me-auto"
              data-bs-dismiss="toast"
              id="toastClose"
              type="button"
            />
          </div>
        </div>
        <div
          class="toast-body"
        />
      </div>
    </div>
    <div
      class="container"
    >
      <form>
        <div
          class="row g-3"
        >
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="name"
              placeholder="Name"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide a valid name.
            </div>
          </div>
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="model"
              placeholder="Model"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide a model name.
            </div>
          </div>
          <div
            class="col-12 mb-3"
          >
            <input
              class="form-control"
              id="image"
              placeholder="Image URL"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide an image URL
            </div>
          </div>
          <div
            class="col-12 mb-3"
          >
            <input
              class="form-control"
              id="description"
              placeholder="Description"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide a description
            </div>
          </div>
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="number_of_seats"
              placeholder="Number of seats"
              required=""
              type="number"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide Number of seats
            </div>
          </div>
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="location"
              placeholder="Location"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide location
            </div>
          </div>
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="fee"
              placeholder="Enter fee"
              required=""
              type="number"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide a fee
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12"
          >
            <button
              class="btn"
              style="background-color: rgb(26, 35, 126); color: white; margin-right: 10px;"
              type="submit"
            >
              Submit
            </button>
            <button
              class="btn btn-secondary"
              type="button"
            >
              Clear
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`renders Aeroplane component correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="col-12 text-center"
    >
      <h3
        class="fw-normal"
      >
         Add a New Aeroplane
      </h3>
    </div>
    <div
      class="toast-container position-fixed top-0 end-0 p-3"
    >
      <div
        aria-atomic="true"
        aria-live="assertive"
        class="toast"
        id="liveToast"
        role="alert"
      >
        <div
          class="toast-header"
        >
          <div
            class="float-sm-start"
          >
            <svg
              class="me-2"
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 4c-4.879 0-9 4.121-9 9s4.121 9 9 9 9-4.121 9-9-4.121-9-9-9zm0 16c-3.794 0-7-3.206-7-7s3.206-7 7-7 7 3.206 7 7-3.206 7-7 7z"
              />
              <path
                d="M13 12V8h-2v6h6v-2zm4.284-8.293 1.412-1.416 3.01 3-1.413 1.417zm-10.586 0-2.99 2.999L2.29 5.294l2.99-3z"
              />
            </svg>
            <i
              class="bi bi-react me-2"
            />
          </div>
          <strong
            class="me-auto"
          >
            Alert
          </strong>
          <div
            class="float-sm-end"
          >
            <button
              aria-label="Close"
              class="btn-close me-auto"
              data-bs-dismiss="toast"
              id="toastClose"
              type="button"
            />
          </div>
        </div>
        <div
          class="toast-body"
        />
      </div>
    </div>
    <div
      class="container"
    >
      <form>
        <div
          class="row g-3"
        >
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="name"
              placeholder="Name"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide a valid name.
            </div>
          </div>
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="model"
              placeholder="Model"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide a model name.
            </div>
          </div>
          <div
            class="col-12 mb-3"
          >
            <input
              class="form-control"
              id="image"
              placeholder="Image URL"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide an image URL
            </div>
          </div>
          <div
            class="col-12 mb-3"
          >
            <input
              class="form-control"
              id="description"
              placeholder="Description"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide a description
            </div>
          </div>
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="number_of_seats"
              placeholder="Number of seats"
              required=""
              type="number"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide Number of seats
            </div>
          </div>
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="location"
              placeholder="Location"
              required=""
              type="text"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide location
            </div>
          </div>
          <div
            class="col-md-6 mb-3"
          >
            <input
              class="form-control"
              id="fee"
              placeholder="Enter fee"
              required=""
              type="number"
              value=""
            />
            <div
              class="invalid-feedback"
            >
              Please provide a fee
            </div>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-12"
          >
            <button
              class="btn"
              style="background-color: rgb(26, 35, 126); color: white; margin-right: 10px;"
              type="submit"
            >
              Submit
            </button>
            <button
              class="btn btn-secondary"
              type="button"
            >
              Clear
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
